### PR TITLE
 the shallow clone command is wrong

### PR DESCRIPTION
**Pull request by talkxin**
the command

"${GIT} ls-remote "${repo}" "${cset}" 2>&1"

It's impossible to find a branch.

### CODE REVIEW

File: buildroot/support/download/git [hunk 1]
```diff
@@ -1,3 +1,8 @@
+#/home/noobs/buildroot/support/download/dl-wrapper -b git -o ./linux-f70eae405b5d75f7c41ea300b9f790656f99a203.tar.gz -H linux//linux.hash -- git://github.com/raspberrypi/linux.git f70eae405b5d75f7c41ea300b9f790656f99a203 linux-f70eae405b5d75f7c41ea300b9f790656f99a203
+#git clone -v --bare git://github.com/raspberrypi/linux.git  linux-f70eae405b5d75f7c41ea300b9f790656f99a203
+#在41行 clone linux内核时 不仅把某一提交版本拉下来并且还把整个仓库都拉了下来，速度太慢
+#在脚本中尝试了浅层克隆，但是由于某种原因失败了，改成了全仓库克隆，克隆了将近3g的仓库
```
Comment (lurch) [inline buildroot/support/download/git]: Ermmm, what?

Comment (talkxin) [reply buildroot/support/download/git]: well,this command
"${GIT} ls-remote "${repo}" "${cset}" 2>&1"
even if this branch exist,it whill never be find.

 this command is effective
 branch="`${GIT} ls-remote "${repo}" | grep -i "${cset}" | awk -F '/' '{print $3}'`"

Comment (talkxin) [reply buildroot/support/download/git]: Actually I made all the references for myself, you can ignore references,thx

Comment (talkxin) [reply buildroot/support/download/git]: I am looking forward the further information

Comment (lurch) [reply buildroot/support/download/git]: Sorry, I was confused by the chinese characters, and didn't realise they were commented-out.

**Comment by lurch** [comment]
I've just been digging into this, and it _does_ work as expected, if given a tag or branch rather than a commit-ID (since you can only shallow-clone tags or branches). So
```
git ls-remote https://github.com/raspberrypi/linux f70eae405b5d75f7c41ea300b9f790656f99a203
```
returns nothing, but
```
git ls-remote https://github.com/raspberrypi/linux raspberrypi-kernel_1.20180417-1
```
returns
```
f70eae405b5d75f7c41ea300b9f790656f99a203	refs/tags/raspberrypi-kernel_1.20180417-1
```

So (AFAICT) the change that you're proposing here only works because `f70eae405b5d75f7c41ea300b9f790656f99a203` just _happens_ to also be tagged as `raspberrypi-kernel_1.20180417-1` - but there's obviously no guarantee that the commit-ID passed to this helper script will always have an associated tag.

And in any case I've just checked the upstream version of this script, and they've actually _removed_ support for shallow clones https://git.busybox.net/buildroot/commit/support/download/git?id=4f54c959dc18cb3ca56e8f54b358fc219352d7bd

**Comment by XECDesign** [comment]
Closing based on @lurch's feedback. Thanks.

**Pull request closed without merging**